syntax = "proto3";

  package chat;

  service ChatService {
    rpc ChatStream (ChatRequest) returns (stream ChatReply);
  }

  message ChatRequest {
    string message = 1;
  }

  message ChatReply {
    string response_chunk = 1;
    string type = 2;
  }